samples/snippets/src/test/java/com/example/spanner/SpannerSampleIT.java

```diff
@@ -93,10 +97,45 @@ public static void setUp() throws Exception {
     dbId = DatabaseId.of(options.getProjectId(), instanceId, databaseId);
     // Delete stale test databases that have been created earlier by this test, but not deleted.
     deleteStaleTestDatabases(instanceId, baseDbId);
-    key = String.format("projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s",
-        options.getProjectId(), keyLocation, keyRing, keyName);
+    key =
+        String.format(
+            "projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s",
+            options.getProjectId(), keyLocation, keyRing, keyName);
+
+    /*
+     * Delete stale instances that have been created earlier by this test but not deleted.
+     * Backups needed to be deleted from the instance first, as the instance can only be
+     * deleted once all backups have been deleted.
+     * */
+    deleteStaleTestEncryptedInstances();
+  }
+
+  /**
+   * Deleting all the test instances with name beginning from 'encrypted-test-' and were created
+   * before 24 hours.
+   *
+   * @throws InterruptedException If Thread.sleep() interrupted
+   */
+  static void deleteStaleTestEncryptedInstances() throws InterruptedException {
+    List<Instance> instances =
+        Lists.newArrayList(
```

Comment:
Instead of creating a new list here using Lists.newArrayList(), we could directly iterate over spanner.getInstanceAdminClient().listInstances(Options.filter("name:encrypted-test-")).iterateAll() as follows:

```
for (Instance instance :
    spanner
        .getInstanceAdminClient()
        .listInstances(Options.filter("name:encrypted-test-"))
        .iterateAll()) {
```